src/shared/features/AppletSettings/ExportDataSetting/ExportDataSetting.tsx

```diff
@@ -1,46 +1,147 @@
-import { useState } from 'react';
+import { useState, useEffect } from 'react';
 import { useTranslation } from 'react-i18next';
 import { Box } from '@mui/material';
+import { addDays } from 'date-fns';
+import { ObjectSchema } from 'yup';
+import { yupResolver } from '@hookform/resolvers/yup';
+import { FormProvider, useForm } from 'react-hook-form';
 
 import { DataExportPopup } from 'modules/Dashboard/features/Respondents/Popups';
 import { Svg } from 'shared/components/Svg';
 import { applet } from 'shared/state';
+import { SelectController } from 'shared/components/FormComponents';
+import { DatePicker } from 'shared/components';
+import { theme, StyledBodyLarge, StyledFlexTopCenter } from 'shared/styles';
 
 import {
   StyledAppletSettingsButton,
   StyledAppletSettingsDescription,
 } from '../AppletSettings.styles';
+import { ExportDataFormValues, ExportDateType } from './ExportDataSettings.types';
+import { exportDataSettingSchema } from './ExportDataSetting.schema';
+import { getDateTypeOptions } from './ExportDataSetting.utils';
 
 export const ExportDataSetting = () => {
   const { t } = useTranslation('app');
   const { result: appletData } = applet.useAppletData() ?? {};
-
   const [dataIsExporting, setDataIsExporting] = useState(false);
-
   const dataTestid = 'applet-settings-export-data-export';
+  const minDate = new Date(appletData?.createdAt ?? '');
+  const maxDate = new Date();
+  const methods = useForm<ExportDataFormValues>({
+    resolver: yupResolver(exportDataSettingSchema() as ObjectSchema<ExportDataFormValues>),
+    defaultValues: {
+      dateType: ExportDateType.AllTime,
+      fromDate: minDate,
+      toDate: maxDate,
+    },
+    mode: 'onSubmit',
+  });
+  const { control, setValue, watch } = methods;
+  const dateType = watch('dateType');
+  const fromDate = watch('fromDate');
+  const toDate = watch('toDate');
+  const hasCustomDate = dateType === ExportDateType.ChooseDates;
+  const commonProps = {
+    minDate,
+    maxDate,
+    control,
+    inputSx: { width: '19rem' },
+  };
+
+  const onCloseCallback = () => {
+    if (toDate < fromDate) {
+      setValue('toDate', addDays(fromDate, 1));
+    }
+  };
+
+  useEffect(() => {
```

Comment:
could you pls move this functionality to Select Controller customChange?
![image](https://github.com/ChildMindInstitute/mindlogger-admin/assets/116339462/57d0146e-beeb-4449-a4a9-b2e6b927f95f)